src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -197,21 +200,22 @@ private static StudyInfos toInfos(StudyEntity entity) {
                 .build();
     }
 
-    private static BasicStudyInfos toBasicInfos(BasicStudyEntity entity) {
+    private static BasicStudyInfos toBasicInfos(StudyCreationRequestEntity entity) {
         return BasicStudyInfos.builder().studyName(entity.getStudyName())
                 .creationDate(ZonedDateTime.ofInstant(entity.getDate().toInstant(ZoneOffset.UTC), ZoneId.of("UTC")))
                 .userId(entity.getUserId())
+                .id(entity.getId())
                 .build();
     }
 
     Flux<StudyInfos> getStudyList(String userId) {
-        return studyRepository.getStudies(userId).map(StudyService::toInfos)
+        return Flux.fromIterable(studyRepository.getStudyList(userId)).map(StudyService::toInfos)
                 .sort(Comparator.comparing(StudyInfos::getCreationDate).reversed());
     }
 
     Flux<BasicStudyInfos> getStudyCreationRequests(String userId) {
-        return studyCreationRequestRepository.getStudyCreationRequests(userId).map(StudyService::toBasicInfos)
-                .sort(Comparator.comparing(BasicStudyInfos::getCreationDate).reversed());
+        return Flux.fromStream(studyCreationRequestRepository.findAllByUserId(userId).stream().map(StudyService::toBasicInfos)
```

Comment:
use the same code as getStudyList, flux.fromIterable and then map on the flux

Reply:
now that the code is the same, you can refactor the duplicated part in a single method

Reply:
I wonder if we should make sure that all database access happens after the mono is subscribed or if it's ok to do it just here. Maybe it would be better to do it after no ?

Reply:
an advantage of doing it after is that we can always easily move some work to a bounded elasticscheduler if needed

Reply:
Yeah, let's try to run all code in the mono/flux, i'll annotate the different places